editor/plugins/visual_shader_editor_plugin.cpp

```diff
@@ -2054,6 +2068,28 @@ void VisualShaderEditor::drop_data_fw(const Point2 &p_point, const Variant &p_da
 			saved_node_pos = p_point;
 			saved_node_pos_dirty = true;
 			_add_node(idx, add_options[idx].sub_func);
+		} else if (d.has("files")) {
+			if (d["files"].get_type() == Variant::POOL_STRING_ARRAY) {
+
+				PoolStringArray arr = d["files"];
+				for (int i = 0; i < arr.size(); i++) {
+
+					String type = ResourceLoader::get_resource_type(arr[i]);
+
+					if (type == "Texture" ||
```

Comment:
On phone right now so can't check, but I think there might be a ClassDB method to check if the type inheritd Texture.

Reply:
Fixed, also improved performance by replacing loop to index insertion in _add_texture_node